testing/square_bracket_accessor_test.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+#
+# square_bracket_accessor_test.py
+# Francois-Michel L'Heureux, 2017-01-27
+# This file is part of MLDB. Copyright 2017 mldb.ai inc. All rights reserved.
+#
+import unittest
+
+mldb = mldb_wrapper.wrap(mldb)  # noqa
+
+class Mldb2105SelectExprSquareBracketColNameSupport(MldbUnitTest):  # noqa
+
+    def test_string_with_match(self):
+        """
+        It doesn't matter that it maches, it's a string.
+        """
+        res = mldb.query("SELECT a['foo'] FROM (SELECT {foo: 123} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', "a['foo']"],
+            ['result', 'foo']
+        ])
+
+    def test_string_no_match(self):
+        res = mldb.query("SELECT a['bar'] FROM (SELECT {foo: 123} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', "a['bar']"],
+            ['result', 'bar']
+        ])
+
+    def test_string_chaining(self):
+        res = mldb.query("SELECT a['f']['b']['z'] FROM (SELECT {f: {b: 123}} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', "a['f']['b']['z']"],
+            ['result', 'z']
+        ])
+
+    def test_string_mixing(self):
+        res = mldb.query("SELECT a.f['b'] FROM (SELECT {f: {b: 123}} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', u'"a.f[\'b\']"'],
+            ['result', 'b']
+        ])
+
+        with self.assertRaises(mldb_wrapper.ResponseException):
+            res = mldb.query("SELECT a['f'].b FROM (SELECT {f: {b: 123}} AS a)")
+
+    def test_string_outter_whitespaces(self):
+        res = mldb.query("SELECT a[ 'foo'   ] FROM (SELECT {foo: 123} AS a)")
+        self.assertEqual(res, [
+            ['_rowName', "a[ 'foo'   ]"],
+            ['result', 'foo']
+        ])
+
+    def test_string_inner_whitespaces(self):
+        res = mldb.query("""
+            SELECT a['f b'] FROM (
+                SELECT parse_json('{"f b" : 123}') AS a
+            )
+                         """)
+        self.assertEqual(res, [
+            ['_rowName', "a['f b']"],
+            ['result', 'f b']
+        ])
+
+    def test_string_utf8(self):
+        res = mldb.query("""
+            SELECT a['utéf8'] FROM (
+                SELECT parse_json('{"utéf8" : 123}') AS a
+            )
+                         """)
+        self.assertEqual(res, [
+            [u'_rowName', u"a['utéf8']"],
+            [u'result', u'utéf8']
+        ])
+
```

Comment:
I think that it's OK that these fail for now; we know at bind time that we're asking for row[null] which isn't valid (will always fail at runtime).  If later on we want to make it so that row[null] = null then we should change that separately.